components/compositing/pinch_zoom.rs

```diff
@@ -34,6 +34,10 @@ impl PinchZoom {
         Scale::new(self.zoom_factor)
     }
 
+    pub(crate) fn resize(&mut self, webview_rect: DeviceRect) {
```

Comment:
Ok I think I understand what you are doing now. But it really makes no sense to be named as `resize` for `PinchZoom` when only changing `unscaled_viewport_size`.

Maybe something like this
```suggestion
    pub(crate) fn resize_unscaled_viewport(&mut self, webview_rect: DeviceRect) {
```